src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -0,0 +1,239 @@
+        _goForwardButton.IsChecked(true);
+        if (_goBackwardButton.IsChecked())
+        {
+            _goBackwardButton.IsChecked(false);
+        }
+
+        // kick off search
+        _searchEventHandler(*this, _textBox.Text());
+    }
+
+    // Method Description:
+    // - Handler for clicking the Case Sensitivity button. This changes
+    //   _isCaseSensitive value
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_CaseSensitivityClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, RoutedEventArgs const& /*e*/)
+    {
+        _isCaseSensitive = !_isCaseSensitive;
+    }
+
+    // Method Description:
+    // - Handler for clicking the close button. This destructs the
+    //   search box object in TermControl
+    // Arguments:
+    // - sender: not used
+    // - e: event data
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_CloseClick(winrt::Windows::Foundation::IInspectable const& /*sender*/, RoutedEventArgs const& e)
+    {
+        _CloseButtonClickedHanlder(*this, e);
+    }
+
+    // Method Description:
+    // - Handler for pressing Enter when focusing on the checkbox.
+    //   This is implented because XAML checkbox does not support
+    //   Enter
+    // Arguments:
+    // - sender: the XAML element responding to the pointer input
+    // - e: event data
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_CheckboxKeyDown(winrt::Windows::Foundation::IInspectable const& sender, Input::KeyRoutedEventArgs const& e)
+    {
+        // Checkbox does not got checked when focused and pressing Enter, we
+        // manually listen to Enter and check here
+        if (e.OriginalKey() == winrt::Windows::System::VirtualKey::Enter)
+        {
+            auto checkbox = sender.try_as<Controls::CheckBox>();
+            if (checkbox.IsChecked().GetBoolean() == true)
+            {
+                checkbox.IsChecked(false);
+            }
+            else
+            {
+                checkbox.IsChecked(true);
+            }
+            e.Handled(true);
+        }
+    }
+
+    // Method Description:
+    // - To avoid Characters input bubbling up to terminal, we implement this handler here,
+    //   simply mark the key input as handled
+    // Arguments:
+    // - sender: not used
+    // - e: event data
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::_CharacterHandler(winrt::Windows::Foundation::IInspectable const& /*sender*/, Input::CharacterReceivedRoutedEventArgs const& e)
+    {
+        e.Handled(true);
+    }
+
+    // Events proxies
+    DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(SearchBoxControl, Search, _searchEventHandler, TerminalControl::SearchBoxControl, winrt::hstring);
+    DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(SearchBoxControl, CloseButtonClicked, _CloseButtonClickedHanlder, TerminalControl::SearchBoxControl, Windows::UI::Xaml::RoutedEventArgs);

```

Comment:
you can use the new `TYPED_EVENT` macro in the .h file and you don't need these two.